benchmark/benchmain/main.go

```diff
@@ -346,16 +353,41 @@ func makeClient(bf stats.Features) (testgrpc.BenchmarkServiceClient, func()) {
 	}
 	lis = nw.Listener(lis)
 	stopper := bm.StartServer(bm.ServerInfo{Type: "protobuf", Listener: lis}, sopts...)
-	conn := bm.NewClientConn("" /* target not used */, opts...)
-	return testgrpc.NewBenchmarkServiceClient(conn), func() {
-		conn.Close()
+	var conns []*grpc.ClientConn
+	var clients []testpb.BenchmarkServiceClient
+	if bf.ShareConnection {
+		conns = []*grpc.ClientConn{bm.NewClientConn("" /* target not used */, opts...)}
+		clients = []testpb.BenchmarkServiceClient{testgrpc.NewBenchmarkServiceClient(conns[0])}
+	} else {
+		conns = make([]*grpc.ClientConn, bf.MaxConcurrentCalls)
+		clients = make([]testpb.BenchmarkServiceClient, bf.MaxConcurrentCalls)
+		for i := 0; i < bf.MaxConcurrentCalls; i++ {
+			conns[i] = bm.NewClientConn("" /* target not used */, opts...)
+			clients[i] = testgrpc.NewBenchmarkServiceClient(conns[i])
+		}
+	}
+
+	return clients, func() {
+		for _, conn := range conns {
+			conn.Close()
+		}
 		stopper()
 	}
 }
 
+func selectClient(clients []testpb.BenchmarkServiceClient, index int) testpb.BenchmarkServiceClient {
+	var tc testpb.BenchmarkServiceClient
+	if len(clients) > 1 {
+		tc = clients[index]
+	} else {
+		tc = clients[0]
+	}
+	return tc
+}
```

Comment:
We can slightly simplify the code and remove this function if we always create `MaxConcurrentCalls` clients, but in case when `ShareConnection == true` all those clients will use a single connection. I didn't implement this options because it modifies the behavior of benchmarks with default setting, which I want to avoid.